sympy/solvers/diophantine/diophantine.py

```diff
@@ -3742,26 +3767,20 @@ def sum_of_four_squares(n):
     See Also
     ========
 
-    sum_of_squares()
+    power_representation :
+        ``sum_of_four_squares(n)`` is one of the solutions output by ``power_representation(n, 2, 4, zeros=True)``
+
     """
+    n = as_int(n)
+    if n < 0:
+        raise ValueError("n should be a non-negative integer")
     if n == 0:
         return (0, 0, 0, 0)
-
-    v = multiplicity(4, n)
-    n //= 4**v
-
-    if n % 8 == 7:
-        d = 2
-        n = n - 4
-    elif n % 8 in (2, 6):
-        d = 1
-        n = n - 1
-    else:
-        d = 0
-
-    x, y, z = sum_of_three_squares(n)
-
-    return _sorted_tuple(2**v*d, 2**v*x, 2**v*y, 2**v*z)
+    n, v = remove(n, 4)
+    v = 1 << v
+    d = 1 if n % 8 == 7 else 0
```

Comment:
```suggestion
    if n % 8 == 7:
        d = 2
        n = n - 4
    elif n % 8 in (2, 6):
        d = 1
        n = n - 1
    else:
        d = 0
```

Reply:
OK, the special 3 and 4 versions are much faster than the generic `sum_of_squares`. But let's keep the non-0 default for 4 squares when possible. Doctest and tests will need updating.

Reply:
These should be listed in the `__all__`, too, since they are public (but not used internally). (?)